sharkfin/population.py

```diff
@@ -481,7 +592,17 @@ def class_stats(self, store=False):
 
                 records.append(record)
 
-        agent_df = pd.DataFrame.from_records(records)
+        agent_dF = pd.DataFrame.from_records(records)
```

Comment:
what is the point of recreating agent_df if it's not what's returned by the method? maybe you meant to update the existing df?

Reply:
It looks like there is a typo here -- `agent_dF` should be `agent_df`

But the more substantive issue is that the schema of the agent_df returned by this method is different from the agent_database that you construct in this class.

`agent_df` probably should be renamed something more descriptive.

But if I am not mistaken, the columns of this agent_df are not included as columns in the agent_database, because they are derived from state variables in the HARK agent objects.

I wouldn't want to add these as columns to the agent_database because it would be troublesome to keep the data in the 'database' in sync with the state values on those agent objects.

This method is really just collecting information needed to do reporting .

Reply:
OK I think also `agent_dF = self.agent_df` so it's the object that is being returned below

Reply:
i see what you're saying now. Thanks, I've fixed that!